File: php/tests/FakeCatalog.php
<?php

declare(strict_types=1);

namespace Tests;

use Supermarket\Model\Product;
use Supermarket\Model\SupermarketCatalog;

class FakeCatalog implements SupermarketCatalog
{
    /**
     * @var array<string, float>
     */
    private array $prices = [];

    /**
     * @var array<string, Product>
     */
    private array $products = [];

    public function addProduct(Product $product, float $price): void
    {
        $this->products[$product->getName()] = $product;
        $this->prices[$product->getName()] = $price;
    }

    public function getUnitPrice(Product $product): float
    {
        return $this->prices[$product->getName()];
    }
}

File: python/tests/fake_catalog.py
from catalog import SupermarketCatalog

from models.product import Product

class FakeCatalog(SupermarketCatalog):
    def __init__(self):
        self.products = {}
        self.prices = {}

    def add_product(self, product: Product, price: int):
        self.products[product.name] = product
        self.prices[product.name] = price

    def unit_price(self, product: Product):
        return self.prices[product.name]

